subsys/bluetooth/mesh/crypto.c

```diff
@@ -292,7 +292,7 @@ int bt_mesh_net_encrypt(const uint8_t key[16], struct net_buf_simple *buf,
 	       mic_len);
 	BT_DBG("PDU (len %u) %s", buf->len, bt_hex(buf->data, buf->len));
 
-	if (IS_ENABLED(CONFIG_BT_MESH_PROXY) && proxy) {
+	if (IS_ENABLED(CONFIG_BT_MESH_GATT) && proxy) {
```

Comment:
This should not be changed, because CONFIG_BT_MESH_PROXY is used by CONFIG_BT_MESH_GATT_PROXY and CONFIG_BT_MESH_PROXY_CLIENT(in the future).